src/plugins/auctionHouseModule/createBid.ts

```diff
@@ -46,9 +94,110 @@ export type CreateBidBuilderParams = Omit<CreateBidInput, 'confirmOptions'> & {
   instructionKey?: string;
 };
 
+export type CreateBidBuilderContext = Omit<CreateBidOutput, 'response'>;
+
 export const createBidBuilder = (
   metaplex: Metaplex,
   params: CreateBidBuilderParams
-): TransactionBuilder => {
-  return TransactionBuilder.make();
+): TransactionBuilder<CreateBidBuilderContext> => {
+  // Data.
+  const auctionHouse = params.auctionHouse;
+  const tokens = params.tokens ?? token(1);
+  const priceBasisPoint = params.price?.basisPoints ?? 0;
+  const price = auctionHouse.isNative
+    ? lamports(priceBasisPoint)
+    : amount(priceBasisPoint, auctionHouse.treasuryMint.currency);
+
+  // Accounts.
+  const wallet = params.wallet ?? (metaplex.identity() as Signer);
+  const authority = params.authority ?? auctionHouse.authorityAddress;
+  const metadata = findMetadataPda(params.mintAccount);
+  const tokenAccount =
+    params.tokenAccount ??
+    findAssociatedTokenAccountPda(params.mintAccount, toPublicKey(wallet));
+  const buyerTradeState = findAuctionHouseTradeStatePda(
+    auctionHouse.address,
+    toPublicKey(wallet),
+    tokenAccount,
+    auctionHouse.treasuryMint.address,
+    params.mintAccount,
+    price.basisPoints,
+    tokens.basisPoints
+  );
+  const escrowPayment = findAuctionHouseBuyerEscrowPda(
+    auctionHouse.address,
+    toPublicKey(wallet)
+  );
+
+  const accounts: BuyInstructionAccounts = {
+    wallet: toPublicKey(wallet),
+    paymentAccount: toPublicKey(wallet),
+    transferAuthority: toPublicKey(wallet),
+    treasuryMint: auctionHouse.treasuryMint.address,
+    tokenAccount,
+    metadata,
+    escrowPaymentAccount: escrowPayment,
+    authority: toPublicKey(authority),
+    auctionHouse: auctionHouse.address,
+    auctionHouseFeeAccount: auctionHouse.feeAccountAddress,
+    buyerTradeState,
+  };
+
+  // Args.
+  const args = {
+    tradeStateBump: buyerTradeState.bump,
+    escrowPaymentBump: escrowPayment.bump,
+    buyerPrice: price.basisPoints,
+    tokenSize: tokens.basisPoints,
+  };
+
+  // Sell Instruction.
+  // ToDo: Add support for the auctioneerAuthority
+  const buyInstruction = createBuyInstruction(accounts, args);
```

Comment:
This needs a lot more work. Ultimately, this instruction can be one of four based on the input:
- bid
- public bid
- bid with an auctioneer
- public bid with an auctioneer